deepchecks/base/dataset.py

```diff
@@ -349,43 +355,48 @@ def n_samples(self) -> int:
         """
         return self.data.shape[0]
 
-    def __len__(self):
+    def __len__(self) -> int:
         """Return number of samples in the member dataframe."""
         return self.n_samples
 
-    def infer_categorical_features(self) -> t.List[Hashable]:
+    @staticmethod
+    def _infer_categorical_features(
```

Comment:
why it's better as static method if it's need so many values from self?

Reply:
I think the next reasons should be followed when deciding if declaring a function as a method instance:
- it represents/modifies object state during its life-cycle
- it is a private method that works with object inner state

`infer_categorical_features` is definitely just a utility function 
that is called only once during dataset initialization, also, 
it neither represents object state nor modifies it

passing arguments explicitly makes it more obvious what does this function do

also, do not you consider it strange that we do not pass arguments to it 
(we use self for this) but we return value from it which is then assigned to the self/